src/ol_infrastructure/components/services/vault.py

```diff
@@ -250,3 +267,232 @@ def __init__(
+            opts,
+        )
+
+        resource_options = ResourceOptions(parent=self).merge(opts)  # type: ignore
+
+        self.pki_intermediate_environment_backend = Mount(
+            f"pki-intermediate-{backend_config.environment_name}",
+            opts=resource_options,
+            path=f"pki-intermediate-{backend_config.environment_name}",
+            type="pki",
+            description=f"Backend to create certs for pki-intermediate-{backend_config.environment_name} backends",
+            max_lease_ttl_seconds=backend_config.max_ttl,
+            default_lease_ttl_seconds=backend_config.default_ttl,
+        )
+
+        # Generate CSR for pki-intermediate-{env} backend
+        self.pki_intermediate_envrionment_cert_request = pkisecret.SecretBackendIntermediateCertRequest(
+            f"pki-intermediate-{backend_config.environment_name}-csr",
+            backend=self.pki_intermediate_environment_backend.id,
+            common_name=f"pki-intermediate-{backend_config.environment_name} Intermediate Authority",
+            type="internal",
+            country=CERTIFICATE_CONFIG["country"],
+            province=CERTIFICATE_CONFIG["state"],
+            locality=CERTIFICATE_CONFIG["city"],
+            organization=CERTIFICATE_CONFIG["organization"],
+            ou=CERTIFICATE_CONFIG["organizational_unit"],
+            postal_code=CERTIFICATE_CONFIG["zip_code"],
+        )
+
+        # Sign genereated CSR for pki-intermediate-{env} backend by pki-intermediate-ca
+        self.pki_intermediate_environment_signed_csr = pkisecret.SecretBackendRootSignIntermediate(
+            f"pki-intermediate-{backend_config.environment_name}-signed-csr",
+            backend="pki-intermediate-ca",
+            common_name=f"pki-intermediate-{backend_config.environment_name} Intermediate Authority",
+            csr=self.pki_intermediate_envrionment_cert_request.csr,
+        )
+
+        self.pki_intermediate_environment_set_signed = (
+            pkisecret.SecretBackendIntermediateSetSigned(
+                f"pki-intermediate-{backend_config.environment_name}-signed-cert",
+                backend=self.pki_intermediate_environment_backend.id,
+                certificate=self.pki_intermediate_environment_signed_csr.certificate,
+            )
+        )
+
+        self.pki_intermediate_environment_config_urls = (
+            pkisecret.SecretBackendConfigUrls(
+                f"pki-intermediate-{backend_config.environment_name}-config-url",
+                backend=self.pki_intermediate_environment_backend.id,
+                crl_distribution_points=[
+                    f"{VAULT_API_URL}/backend_confg.environment_name/crl"
+                ],
+                issuing_certificates=[
+                    f"{VAULT_API_URL}/backend_confg.environment_name/ca"
+                ],
+            )
+        )
+
+        self.register_outputs({})
+
+
+class OLVaultPKIIntermediateRoleConfig(BaseModel):
+    pki_intermediate_backend: Text
```

Comment:
Should we set this to be an instance of the actual backend object to create a dependency chain from the role to the PKI mount? That will allow us to ensure that we are pointing at a valid backend rather than relying on a stringly typed field.